src/preprocess/base.py

```diff
@@ -48,12 +48,24 @@ def __init__(self, data_folder: Path = Path('data')) -> None:
             print('A dataset attribute must be added for '
                   'the interim and out directories to be created')
 
-    def get_filepaths(self, folder: str = 'raw') -> List[Path]:
+    def get_filepaths(self, folder: str = 'raw',
+                      years: Optional[List[int]] = None) -> List[Path]:
         if folder == 'raw':
             target_folder = self.raw_folder / self.dataset
         else:
             target_folder = self.interim
-        outfiles = list(target_folder.glob('**/*.nc'))
+
+        if years is not None:
+            assert int(list(target_folder.glob('**'))[0])  # type: ignore
+            outfiles = [
+                f for f in target_folder.glob('**/*.nc')
+                if int(f.parents[0]) in years  # type: ignore
```

Comment:
This assumes the files are exported to a `year/name.nc` format - this is not necessarily the case (for instance, the ESA CCI exporter doesn't do that). 

What was the motivation for this change?

Reply:
to only preprocess a subset of the years (if that option is specifieD).

speeds the preprocessing up when it's being tested on servers etc.

plus means that in future if only want to do a subset of the years you can preprocess them. but should probably be better documented or done elsewhere